drivers/serial/uart_rpi_pico.c

```diff
@@ -302,6 +309,7 @@ static const struct uart_driver_api uart_rpi_driver_api = {
 		.uart_dev = uart##idx,						\
 		.uart_regs = (uart_hw_t *)uart##idx,				\
 		.pcfg = PINCTRL_DT_INST_DEV_CONFIG_GET(idx),			\
+		.reset = RESET_DT_CONFIG_INIT(idx, id),				\
```

Comment:
to keep consistency with the other subsystems, use `RESET_DT_INST_DEV_CONFIG_GET`.

Besides that LGTM.